src/overlays/actors/ovl_Demo_Gj/z_demo_gj.c

```diff
@@ -96,241 +84,1326 @@ static ColliderCylinderInitType1 sCylinderInit = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Gj/DemoGj_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Gj/DemoGj_Init.s")
+typedef void (*DemoGjActionFunc)(DemoGj*, GlobalContext*);
+
+// bits 11-15
+s32 DemoGj_GetCollectibleType(DemoGj* this) {
+    s32 params = this->dyna.actor.params >> 0xB;
+
+    return params & 0x1F;
+}
+
+// bits 8-10
+s32 DemoGj_GetCollectibleAmount(DemoGj* this) {
+    s32 params = this->dyna.actor.params >> 0x8;
+
+    return params & 7;
+}
+
+// bits 0-7
+s32 DemoGj_GetType(DemoGj* this) {
+    s32 params = this->dyna.actor.params;
+
+    return params & 0xFF;
+}
+
+void DemoGj_Cylinder_Init(DemoGj* this, GlobalContext* globalCtx, ColliderCylinder* cylinder,
+                          ColliderCylinderInitType1* cylinderInit) {
+    Collider_InitCylinder(globalCtx, cylinder);
+    Collider_SetCylinderType1(globalCtx, cylinder, &this->dyna.actor, cylinderInit);
+}
+
+s32 DemoGj_Cylinder_HasExploded(DemoGj* this, GlobalContext* globalCtx, ColliderCylinder* cylinder) {
+    if (Actor_GetCollidedExplosive(globalCtx, &cylinder->base) != NULL) {
+        return 1; // return true; (?)
+    }
+    return 0; // return false; (?)
+}
+
+void DemoGj_Cylinder_Destroy(DemoGj* this, GlobalContext* globalCtx) {
+    switch (DemoGj_GetType(this)) {
+        case DemoGj_Type_16:
+            Collider_DestroyCylinder(globalCtx, &this->cylinders[0]);
+            Collider_DestroyCylinder(globalCtx, &this->cylinders[1]);
+            Collider_DestroyCylinder(globalCtx, &this->cylinders[2]);
+            break;
+
+        case DemoGj_Type_17:
+            Collider_DestroyCylinder(globalCtx, &this->cylinders[0]);
+            Collider_DestroyCylinder(globalCtx, &this->cylinders[1]);
+            Collider_DestroyCylinder(globalCtx, &this->cylinders[2]);
+            return;
+
+        case DemoGj_Type_22:
+            Collider_DestroyCylinder(globalCtx, &this->cylinders[0]);
+            return;
+
+        default:
+            break;
+    }
+}
+
+void DemoGj_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    DemoGj* this = THIS;
+
+    DemoGj_Cylinder_Destroy(this, globalCtx);
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
+
+void DemoGj_PlayExplosionSfx(DemoGj* this, GlobalContext* globalCtx) {
+    Audio_PlaySoundAtPosition(globalCtx, &this->dyna.actor.world.pos, 50, NA_SE_EV_GRAVE_EXPLOSION);
+}
+
+static Vec3f sVelocity = { 0.0f, 6.0f, 0.0f };
+static Vec3f sAccel = { 0.0f, 0.0f, 0.0f };
+static Color_RGBA8 sPrimColor = { 0, 0, 0, 0 };
+static Color_RGBA8 sEnvColor = { 0, 0, 0, 0 };
+
+void func_80978AFC(GlobalContext* globalCtx, Vec3f* pos, f32 arg2) {
+    f32 temp = arg2 * 0.2f;
+    if (1) {
+        // this function just calls EffectSsDust_Spawn() with drawflags=1 and updateMode=0
+        func_800283D4(globalCtx, pos, &sVelocity, &sAccel, &sPrimColor, &sEnvColor, temp * Rand_ZeroOne() + arg2, 0xF,
+                      0x5A);
+    }
```

Comment:
a few things here. since this data is local to the function you can move it inside and remove the `s` prefix.
I also removed the comment and changed the hex args to decimal since they make more sense that way.
hopefully that if(1) can stay on its own line without wrapping stuff. you may need to shuffle it around a bit to find the right place for it
```suggestion

void func_80978AFC(GlobalContext* globalCtx, Vec3f* pos, f32 arg2) {
    static Vec3f velocity = { 0.0f, 6.0f, 0.0f };
    static Vec3f accel = { 0.0f, 0.0f, 0.0f };
    static Color_RGBA8 primColor = { 0, 0, 0, 0 };
    static Color_RGBA8 envColor = { 0, 0, 0, 0 };
    f32 temp = arg2 * 0.2f;
    
    if (1) { }
    func_800283D4(globalCtx, pos, &sVelocity, &sAccel, &sPrimColor, &sEnvColor, temp * Rand_ZeroOne()+ arg2, 15, 90);
    
```